contrib/pg_tde/src/access/pg_tde_xlog_smgr.c

```diff
@@ -349,20 +424,22 @@ tdeheap_xlog_seg_read(int fd, void *buf, size_t count, off_t offset,
  * [De]Crypt buffer if needed based on provided segment offset, number and TLI
  */
 void
-TDEXLogCryptBuffer(void *buf, size_t count, off_t offset,
+TDEXLogCryptBuffer(const void *buf, void *out_buf, size_t count, off_t offset,
 				   TimeLineID tli, XLogSegNo segno, int segSize)
 {
 	WALKeyCacheRec *keys = pg_tde_get_wal_cache_keys();
 	XLogRecPtr	write_key_lsn;
 	WalLocation data_end = {.tli = tli};
 	WalLocation data_start = {.tli = tli};
 
-	if (!keys)
+	if (keys == NULL)
 	{
 		WalLocation start = {.tli = 1,.lsn = 0};
 
 		/* cache is empty, try to read keys from disk */
-		keys = pg_tde_fetch_wal_keys(start);
+		pg_tde_fetch_wal_keys(start);
+
+		keys = pg_tde_get_wal_cache_keys();
```

Comment:
If the cache is empty why can't we use the return value of `pg_tde_fetch_wal_keys()`? Maybe we should actually stop returning something from it and always do two fucntion calls?